### PR TITLE
Unable to figure out the mocking for the sms, otherwise all complete

### PR DESCRIPTION
**Pull request by TinyGobby**

### CODE REVIEW

File: lib/order.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require 'dotenv/load'
+require_relative 'menu'
+require_relative 'sms'
+
+class Order
+  attr_reader :current_total
+
+  def initialize
```
Comment (EdwardAndress) [inline lib/order.rb]: Did you try injecting the Menu and Sms classes here?

File: lib/order.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  def add(list, customer_number)
+    add_up_given(list)
+    fail 'Incorrect total' if incorrect?(list[:total])
+    confirm_order(customer_number)
+  end
+
+  private
+
+  def add_up_given(list)
+    list.each do |dish, amount|
```
Comment (EdwardAndress) [inline lib/order.rb]: Have you come across `Enumerable#reduce` or `Enumerable#inject`? Neither are particularly readable (to me, at least!) but either could be used to turn this into a one-liner.

File: lib/order.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      @current_total += @menu.prices[dish].to_f * amount.to_f
+    end
+  end
+
+  def incorrect?(given_total)
```
Comment (EdwardAndress) [inline lib/order.rb]: Who calculates `given_total`?

File: lib/sms.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+class Sms
```
Comment (EdwardAndress) [inline lib/sms.rb]: Good decision to create a new class here

File: lib/takeaway.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+require_relative 'menu'
+require_relative 'order'
+
+class Takeaway  
+  def initialize
```
Comment (EdwardAndress) [inline lib/takeaway.rb]: Dependency injection could also be useful here!

File: lib/takeaway.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    @menu = Menu.new
```
Comment (EdwardAndress) [inline lib/takeaway.rb]: Why do both Takeaway and Order need an instance of Menu?  I assume that it's so Order can check that a given item is on the Menu but... it's another point of coupling in the program.

File: spec/order_spec.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  let(:failing_order) { { sushi: 2, burger: 1, total: 3 } }
+  let(:number) { double(:customer_phone_number) }
+  
+  describe 'initialize' do
+    it 'creates a new order' do
+      expect(subject.current_total).to eq 0
+    end
+  end
+
+  describe '#add' do
```
Comment (EdwardAndress) [inline spec/order_spec.rb]: Are these commented out to prevent texts from getting sent?

**Comment by EdwardAndress** [comment]
Generally a good solution.  How would you deal with multiple Takeaways or changing Menus though?  At the moment you'd need to load the files, write some code to create the objects, then edit the files, load them again and create some more objects.  Dependency injection is the key to avoiding this pattern and making this code easier to change.  It would also allow you to avoid sending texts when you run the tests (because you could use mocking more easily).

File: lib/order.rb (hunk 2 above)
Comment (TinyGobby) [reply lib/order.rb]: I've not been able to figure out `Enumerable#inject` (seems to be the right one as I'd need to pass a block) as I need to be able to filter out one of the key/value pairs passed by the hash. I did however change it to: 
`list.each { |dish, amount| @current_total += @menu.prices[dish].to_f * amount }`

File: lib/order.rb (hunk 1 above)
Comment (TinyGobby) [reply lib/order.rb]: Thanks! I hadn't thought about that before.

**Pull request closed without merging**